src/android/android-uvc.cpp

```diff
@@ -0,0 +1,598 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+#ifdef RS2_USE_ANDROID_BACKEND
+
+
+#include "android-uvc.h"
+#include "../types.h"
+#include "libuvc/utlist.h"
+#include "usb_host/device_watcher.h"
+#include "../libuvc/utlist.h"
+
+
+#include <limits>
+
+
+#define MAX_PINS 5
+#define INTERRUPT_BUFFER_SIZE 0x400
+
+namespace librealuvc {
+    namespace platform {
+        // we are using standard fourcc codes to represent formats, while MF is using GUIDs
+        // occasionally there is a mismatch between the code and the guid data
+        const std::unordered_map<uint32_t, uint32_t> fourcc_map = {
+                {0x59382020, 0x47524559},    /* 'GREY' from 'Y8  ' */
+                {0x52573130, 0x70524141},    /* 'pRAA' from 'RW10'.*/
+                {0x32000000, 0x47524559},    /* 'GREY' from 'L8  ' */
+                {0x50000000, 0x5a313620},    /* 'Z16'  from 'D16 ' */
+                {0x52415738, 0x47524559},    /* 'GREY' from 'RAW8' */
+                {0x52573136, 0x42595232}     /* 'RW16' from 'BYR2' */
+        };
+
+        bool android_uvc_device::is_connected(const uvc_device_info &info) {
+            auto result = true;
+
+            //TODO: fix this
```

Comment:
We ought to fix this. Maybe we could check if the the passed in `uvc_device_info` matches any present in the returned vector when we call `query_uvc_devices` on `_backend`?

Reply:
Hm yeah. Had a look around how this is called, and I think it might be best to leave this is as it is for now

The call hierarchy shows this being called in a few places, including the `VideoCapture::read()` call (via `android_uvc_device::stream_on()`, followed by `android_uvc_device::check_connection()`). This means essentially we'll end up enumerating the USB devices on every frame which could end up costly.

Once we get a bit more clever with how we add/remove devices then we can do this. What do you think?

Reply:
Oof yeah I see what you mean. I suppose the other option would be to make this class own a reference to a `android_device_watcher` and put a callback in to `start()` that would cache the connected and disconnected devices when when they changed, so this check would be quick.
More than anything I'm uncomfortable with this TODO given that it's expected functionality in new code :/

Reply:
Yeah that would be nice, but...
1) `android_device_watcher` is only pretending to be a class right now. It has no members and all its functions do is work with static variables declared in the source file. the `start()` function you mention writes the callback you give it to a single variable, meaning that we can only have one single watcher 😟 
2) We don't really do anything if a device is unplugged. We don't have the functionality right now (though I have an idea in my head). We could remove devices from the cache every time we query it, but I don't think there's a point to do it right now. Please see https://github.com/leapmotion/librealuvc/pull/14#discussion_r591351972